package/network/services/dnsmasq/files/dnsmasq.init

```diff
@@ -743,6 +743,21 @@ dhcp_mx_add() {
 	xappend "--mx-host=$service"
 }
 
+dhcp_txt_add() {
+	local cfg="$1"
+	local strs=
+	config_get name "$cfg" name
```

Comment:
Please don't derive name from the config. Add a name field to the record, since this is the paradigm that luci uses. It prevents dnsmasq from crashing during startup when encountering anonymous records.

Reply:
I notice now that I should have made `name` be local, otherwise I don't understand. I took `dhcp_host_add()` as an example, which uses `config_get name "$cfg" name`.

If the issue is the option label in the UCI config file needs to be different for each kind of record, can I just make it `tname` (like `cname`)?

Reply:
Sorry, you had it right - had to double-check the `config_get` function. 

Yes, this way is fine.